backend/src/types/types.ts

```diff
@@ -55,6 +60,38 @@ export interface AccessRequestUserPermissions {
   deleteAccessRequest: PermissionDetail
 }
 
+export const FederationState = {
+  DISABLED: 'disabled',
+  READ_ONLY: 'readOnly',
+  ENABLED: 'enabled',
+} as const
+
+export type FederationStateKeys = (typeof FederationState)[keyof typeof FederationState]
+
+export interface RemoteFederationConfig {
+  state: FederationStateKeys
+  baseUrl: string
+  label: string
+  kind: PeerKindKeys
+  proxy: string
+  httpConfig: ProxyAgentOptions
+}
+
+export type SystemStatus = {
+  code: number
+  ping?: string
+  federation?: {
```

Comment:
Very very nit-picky of me, but can we split `federation` into its own type? Purely because the rest of these seem to be split up.